src/utils/object_pool.hpp

```diff
@@ -80,6 +81,26 @@ class ObjectPool {
   void AddFreeObjectToPool(const T &in) { available_.push(in); }
   void AddFreeObjectToPool(T &&in) { available_.emplace(in); }
 
+  // JMM: Note that calling this while, e.g., comm buffers still exist
+  // will make the memory a comm buffer points to
+  // invalid. Fortunately, comm buffers always call `Allocate`
+  // beforehand.
```

Comment:
I don't think this is true. Calling `ObjectPool::Clear()` will just remove all of the `Kokkos::views` from the object pool. If they are still referenced elsewhere (e.g. in `Mesh::boundary_comm_map`), the memory should still be valid since the view reference count never dropped to zero. As long as the `ObjectPool` object hasn't been destroyed, eventually calling the destructor of the object that has been released from the pool should still work fine, since a call to `IsValid` will return `false` and the destructor won't do anything to the object pool it points at.